client/rest-high-level/src/test/java/org/elasticsearch/client/MachineLearningIT.java

```diff
@@ -2803,6 +2813,19 @@ public void testUpdateModelSnapshot() throws Exception {
             getModelSnapshotsResponse2.snapshots().get(0).getDescription());
     }
 
+    public void testUpgradeJobSnapshot() throws Exception {
+        String jobId = "test-upgrade-model-snapshot";
+        String snapshotId = "1541587919";
+
+        createModelSnapshot(jobId, snapshotId, Version.V_7_0_0);
+        //TODO add a true state from the past somehow
```

Comment:
I am not 100% how to do this. Right now this test effectively just checks that the parameters are parsed and sent as the resulting error indicates that we at least tried to load the model snapshot.

The solution might be to get an actual snapshot, and then manually updating the doc so that the min_version is old.

Reply:
> The solution might be to get an actual snapshot, and then manually updating the doc so that the min_version is old.

Yes, in terms of testing the infrastructure that would be a good way.  Run a simple job like farequote, update the model snapshot document after the job is closed to have a `min_version` from the previous major, then upgrade it.  Not sure this needs to be done in the HRLC tests though - for such a complex test the native multi node tests seems like the single place to do it.  I am happy to leave this test as-is, just testing the parameter passing.

In terms of testing actual upgrade, it could be done in the BWC tests.  We could have a BWC test (Java, not YAML) that does nothing when the old cluster is on the same major, but when the old cluster is on a different major it opens/runs/closes a job in the old cluster, then upgrades its model snapshot in the fully upgraded cluster (and does nothing in the mixed cluster).

Reply:
yeah, I have a BWC test class covering this case.